src/pages/OrderPage.tsx

```diff
@@ -167,20 +191,59 @@ const OrderPage = () => {
     reValidateMode: 'onChange',
   });
 
-  const onSubmit = (data: FormValues) => {
+  useEffect(() => {
+    if (user?.name) {
+      setValue('sender', user.name);
+    }
+  }, [user, setValue]);
+
+  const onSubmit = async (form: FormValues) => {
     if (!product) return;
 
+    if (recipients.length === 0) {
+      toast.error('받는 사람이 없습니다.');
+      return;
+    }
+
     alert(
       `주문이 완료되었습니다.\n` +
         `상품명: ${product.name}\n` +
         `구매 수량: ${recipients.reduce((sum, r) => sum + r.quantity, 0)}\n` +
-        `발신자 이름: ${data.sender}\n` +
-        `메시지: ${data.message}`
+        `발신자 이름: ${form.sender}\n` +
+        `메시지: ${form.message}`
     );
-    navigate(ROUTE.MAIN);
+
+    try {
+      const payload: OrderRequest = {
+        productId: product.id,
+        message: form.message,
+        messageCardId: String(selectedCardId),
+        ordererName: form.sender,
+        receivers: recipients.map((r) => ({
+          name: r.name,
+          phoneNumber: r.phone,
+          quantity: r.quantity,
+        })),
+      };
+
+      await postOrder(payload);
+      navigate(ROUTE.MAIN);
+    } catch (error: unknown) {
+      const err = error as AxiosError<{ data: { message: string } }>;
```

Comment:
이렇게 써도 틀린 방법은 아니지만 아래처럼 사용해보면 어떨까요?
```suggestion
    } catch (error) {
      if (error instanceof axios.AxiosError) {
```

Reply:
error가 unknown타입일때 런타임 오류가 발생할수도 있어서 instanceof를 써서 AxiosError인지 확실히 판단한 다음에 접근하는 것이 좋은 것 같습니다. 피드백 반영하겠습니다 감사합니다!